src/lib/table/table-data-source.ts

```diff
@@ -97,22 +99,51 @@ export class MatTableDataSource<T> implements DataSource<T> {
    */
   sortingDataAccessor: ((data: T, sortHeaderId: string) => string|number) =
       (data: T, sortHeaderId: string): string|number => {
-    const value: any = data[sortHeaderId];
+    let value: any = data[sortHeaderId];
+    return isNumberValue(value) ? Number(value) : value;
+  }
 
-    // If the value is a string and only whitespace, return the value.
-    // Otherwise +value will convert it to 0.
-    if (typeof value === 'string' && !value.trim()) {
-      return value;
-    }
+  /**
+   * Returns a sorted array of the provided data according to the state of the MatSort. Called
+   * after changes are made to the filtered data or when sort changes are emitted from MatSort.
+   * By default, the function retrieves the active sort and its direction and compares data
+   * by retrieving data using the sortingDataAccessor. May be overridden for a custom implementation
+   * of data ordering.
+   * @param data The array of data that should be sorted.
+   * @param sort The connected MatSort that holds the current sort state.
+   */
+  sortPredicate: ((data: T[], sort: MatSort) => T[]) = (data: T[], sort: MatSort): T[] => {
```

Comment:
A "predicate" is a function that returns `true` or `false`, so this isn't quite the right name.

Reply:
Any suggestions? Just `sort`?

Reply:
Sgtm

Reply:
Ugh duh, `sort` is an input for the `MatSort`. Hmm, what about...`sortData`? Kinda fits with the function using `sortDataAccessor`